recipes/_attributes.rb

```diff
@@ -2,17 +2,30 @@
 # Recipe:: _attributes
 # Recipe to set derived attributes
 
-version_tag = "zookeeper-#{node['apache_zookeeper']['version']}"
-node.default['apache_zookeeper']['binary_url'] = ::File.join(node['apache_zookeeper']['mirror'],
-                  "#{version_tag}/#{version_tag}.tar.gz")
 
-base_dir = ::File.join(node['apache_zookeeper']['install_dir'], 'current')
-local_state_dir = node['apache_zookeeper']['local_state_dir']
+case node['apache_zookeeper']['install']['type']
+when 'source'
+  version_tag = "zookeeper-#{node['apache_zookeeper']['version']}"
+  node.default['apache_zookeeper']['binary_url'] = ::File.join(node['apache_zookeeper']['mirror'],
+                    "#{version_tag}/#{version_tag}.tar.gz")
+
+  base_dir = ::File.join(node['apache_zookeeper']['install_dir'], 'current')
+  local_state_dir = node['apache_zookeeper']['local_state_dir']
+
+  node.default['apache_zookeeper']['bin_dir'] = ::File.join(base_dir, 'bin')
+  node.default['apache_zookeeper']['data_dir'] = ::File.join(local_state_dir, 'data')
+  node.default['apache_zookeeper']['log_dir'] = ::File.join(local_state_dir, 'log')
+when 'package'
+  base_dir = node['apache_zookeeper']['install_dir'].sub 'opt', 'etc'
```

Comment:
Would it make more sense just to set this to `/etc/zookeeper` instead of replacing?